File: src/Python/Parcial2/ActividadP2A4/empleado.py
from abc import ABC, abstractmethod

class Empleado(ABC):
    def __init__(self, nombre, apellido, nss):
        self.primerNombre = nombre
        self.apellidoPaterno = apellido
        self.numeroSeguroSocial = nss

    def establecerPrimerNombre(self, nombre):
        self.primerNombre = nombre

    def obtenerPrimerNombre(self):
        return self.primerNombre

    def establecerApellidoPaterno(self, apellido):
        self.apellidoPaterno = apellido

    def obtenerApellidoPaterno(self):
        return self.apellidoPaterno

    def establecerNumeroSeguroSocial(self, nss):
        self.numeroSeguroSocial = nss

    def obtenerNumeroSeguroSocial(self):
        return self.numeroSeguroSocial

    def __str__(self):
        return f"{self.obtenerPrimerNombre()} {self.obtenerApellidoPaterno()}\nnumero de seguro social: {self.obtenerNumeroSeguroSocial()}"

    @abstractmethod
    def ingresos(self):
        pass

File: src/C#/Parcial2/ActividadP2A4/Empleado.cs
using System;

public abstract class Empleado
{
    private string primerNombre;
    private string apellidoPaterno;
    private string numeroSeguroSocial;

    public Empleado(string nombre, string apellido, string nss)
    {
        primerNombre = nombre;
        apellidoPaterno = apellido;
        numeroSeguroSocial = nss;
    }

    public void EstablecerPrimerNombre(string nombre) { primerNombre = nombre; }
    public string ObtenerPrimerNombre() { return primerNombre; }
    public void EstablecerApellidoPaterno(string apellido) { apellidoPaterno = apellido; }
    public string ObtenerApellidoPaterno() { return apellidoPaterno; }
    public void EstablecerNumeroSeguroSocial(string nss) { numeroSeguroSocial = nss; }
    public string ObtenerNumeroSeguroSocial() { return numeroSeguroSocial; }

    public override string ToString()
    {
        return ObtenerPrimerNombre() + " " + ObtenerApellidoPaterno() + "\nnumero de seguro social: " + ObtenerNumeroSeguroSocial();
    }

    public abstract double Ingresos();
}
